src/java/za/co/mip/ablduck/ABLDuckClassVisitor.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck;
+
+import java.util.ArrayList;
+
+import org.apache.tools.ant.Task;
+
+import java.io.IOException;
+
+import com.openedge.core.runtime.IPropath;
+
+import eu.rssw.pct.oedoc.ClassDocumentationVisitor;
+
+import eu.rssw.rcode.ClassCompilationUnit;
+import eu.rssw.rcode.Method;
+import eu.rssw.rcode.Property;
+import eu.rssw.rcode.Parameter;
+import eu.rssw.rcode.AccessModifier;
+
+import za.co.mip.ablduck.models.SourceJSObject;
+import za.co.mip.ablduck.models.generic.DeprecatedObject;
+import za.co.mip.ablduck.models.source.MemberObject;
+import za.co.mip.ablduck.models.source.ParameterObject;
+import za.co.mip.ablduck.utilities.CommentParser;
+import za.co.mip.ablduck.utilities.CommentParseResult;
+
+public class ABLDuckClassVisitor extends ClassDocumentationVisitor {
+    private CommentParser comments;
+
+    public ABLDuckClassVisitor(IPropath propath, Task ablduck) {
+        super(propath);
+
+        this.comments = new CommentParser(ablduck);
+    }
+
+    public SourceJSObject getJSObject() throws IOException {
+        SourceJSObject js = new SourceJSObject();
+        ClassCompilationUnit cu = getClassCompilationUnit();
+        String fullClassName;
+
+        if (cu.packageName != null)
+            fullClassName = cu.packageName + "." + cu.className;
+        else
+            fullClassName = cu.className;
+
+        js.id = "class-" + fullClassName;
+        js.tagname = "class";
+        js.name = fullClassName;
+        js.shortname = cu.className;
+        js.classIcon = "class";
+
+        if (cu.inherits != null)
+            js.ext = cu.inherits;
+
+        String c = null;
+        if (cu.classComment.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1155)